plugins/modules/systemd_info.py

```diff
@@ -0,0 +1,367 @@
+        "loadstate": "loaded",
+        "name": "systemd-journald.socket",
+        "substate": "running",
+        "unitfilepreset": "disabled",
+        "unitfilestate": "static"
+    }
+  }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_command(module, cmd):
+    rc, stdout, stderr = module.run_command(cmd, check_rc=True)
+    return stdout.strip()
+
+
+def parse_show_output(output):
+    result = {}
+    for line in output.splitlines():
+        if "=" in line:
+            key, val = line.split("=", 1)
+            key = key.lower()
+            if key not in result:
+                result[key] = val
+    return result
+
+
+def get_unit_properties(module, systemctl_bin, unit, prop_list):
+    cmd = [systemctl_bin, "show", "-p", ",".join(prop_list), "--", unit]
+    output = run_command(module, cmd)
+    return parse_show_output(output)
+
+
+def determine_category(unit):
+    if unit.endswith('.service'):
+        return 'service'
+    elif unit.endswith('.target'):
+        return 'target'
+    elif unit.endswith('.socket'):
+        return 'socket'
+    elif unit.endswith('.mount'):
+        return 'mount'
+    else:
+        return None
+
+
+def extract_unit_properties(unit_data, prop_list):
+    lowerprop = [x.lower() for x in prop_list]
+    extracted = {
+        prop: unit_data[prop] for prop in lowerprop if prop in unit_data
+    }
+    return extracted
+
+
+def unit_exists(module, systemctl_bin, unit):
+    cmd = [systemctl_bin, "show", "-p", "LoadState", "--", unit]
+    rc, stdout, stderr = module.run_command(cmd)
+    return (rc == 0)
+
+
+def validate_unit_and_properties(module, systemctl_bin, unit, extra_properties):
+    cmd = [systemctl_bin, "show", "-p", "LoadState", "--", unit]
+
+    output = run_command(module, cmd)
+    if "loadstate=not-found" in output.lower():
+        module.fail_json(msg="Unit '{0}' does not exist or is inaccessible.".format(unit))
+
+    if extra_properties:
+        unit_data = get_unit_properties(module, systemctl_bin, unit, extra_properties)
+        missing_props = [prop for prop in extra_properties if prop.lower() not in unit_data]
+        if missing_props:
+            module.fail_json(msg="The following properties do not exist for unit '{0}': {1}".format(unit, ", ".join(missing_props)))
+
+
+def main():
+    module_args = dict(
+        unitname=dict(type='list', elements='str', default=[]),
+        extra_properties=dict(type='list', elements='str', default=[])
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    try:
+        systemctl_bin = module.get_bin_path('systemctl', required=True)
+    except Exception as e:
+        module.fail_json(msg="systemctl binary not found: {0}".format(str(e)))
+
+    try:
+        run_command(module, [systemctl_bin, '--version'])
+    except Exception as e:
+        module.fail_json(msg="Failed to run systemctl: {0}".format(str(e)))
```

Comment:
`AnsibleModule.run_command` by default also does not raise exceptions, but calls `fail_json`.
```suggestion
    run_command(module, [systemctl_bin, '--version'])
```

Reply:
Removed try block